src/components/dialogs/voltage-level/modification/voltage-level-modification-dialog.js

```diff
@@ -0,0 +1,195 @@
+
+    const methods = useForm({
+        defaultValues: emptyFormData,
+        resolver: yupResolver(schema),
+    });
+
+    const { reset, setValue } = methods;
+
+    useEffect(() => {
+        if (editData) {
+            reset(
+                {
+                    [EQUIPMENT_ID]: editData?.equipmentId ?? '',
+                    [EQUIPMENT_NAME]: editData?.equipmentName?.value ?? '',
+                    [SUBSTATION_ID]: editData?.substationId?.value ?? null,
+                    [NOMINAL_VOLTAGE]: editData?.nominalVoltage?.value ?? null,
+                    [LOW_VOLTAGE_LIMIT]:
+                        editData?.lowVoltageLimit?.value ?? null,
+                    [HIGH_VOLTAGE_LIMIT]:
+                        editData?.highVoltageLimit?.value ?? null,
+                    [LOW_SHORT_CIRCUIT_CURRENT_LIMIT]:
+                        editData?.ipMin?.value ?? null,
+                    [HIGH_SHORT_CIRCUIT_CURRENT_LIMIT]:
+                        editData?.ipMax?.value ?? null,
+                },
+                { keepDefaultValues: true }
+            );
+        }
+    }, [editData, reset]);
+
+    const onEquipmentIdChange = useCallback(
+        (equipmentId) => {
+            if (equipmentId) {
+                setDataFetchStatus(FetchStatus.RUNNING);
+                fetchVoltageLevel(studyUuid, currentNodeUuid, equipmentId)
+                    .then((voltageLevel) => {
+                        if (voltageLevel) {
+                            setVoltageLevelInfos(voltageLevel);
+                            setDataFetchStatus(FetchStatus.SUCCEED);
+
+                            //TODO We display the previous value of the substation id in the substation field because we can't change it
+                            // To be removed when it is possible to change the substation of a voltage level in the backend (Powsybl)
+                            setValue(SUBSTATION_ID, voltageLevel?.substationId);
+                        }
+                    })
+                    .catch(() => {
+                        setVoltageLevelInfos(null);
+                        setDataFetchStatus(FetchStatus.FAILED);
+                    });
+            } else {
+                setVoltageLevelInfos(null);
+                reset(emptyFormData, { keepDefaultValues: true });
+            }
+        },
+        [studyUuid, currentNodeUuid, reset, setValue]
+    );
+
+    const onSubmit = useCallback(
+        (voltageLevel) => {
+            modifyVoltageLevel(
+                studyUuid,
+                currentNodeUuid,
+                voltageLevel[EQUIPMENT_ID],
+                voltageLevel[EQUIPMENT_NAME],
+                voltageLevel[NOMINAL_VOLTAGE],
+                voltageLevel[LOW_VOLTAGE_LIMIT],
+                voltageLevel[HIGH_VOLTAGE_LIMIT],
+                voltageLevel[LOW_SHORT_CIRCUIT_CURRENT_LIMIT],
+                voltageLevel[HIGH_SHORT_CIRCUIT_CURRENT_LIMIT],
+                !!editData,
```

Comment:
Maybe add a comment to help understand `!!editData`'s role ? Isn't it confusing to also have `isUpdate` in another part of this file ?